Adding temporal_sharing code and other latest chenges into xdna repo

saifuddin-xilinx opened the pull request.

saifuddin-xilinx committed:
Adding temporal_sharing code and other latest chenges into xdna repo

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

chvamshi-xilinx commented on src/driver/amdxdna/amdxdna_of_drv.c:
```diff
@@ -80,12 +84,34 @@ static int amdxdna_of_probe(struct platform_device *pdev)
+	}
+
+	if (xdna->dev_handle)
+		xdna->dev_handle->xrs_hdl = xrsm_init(&xrs_cfg);
```
add a new line

chvamshi-xilinx commented on src/driver/amdxdna/ve2_debug.c:
```diff
@@ -20,7 +20,8 @@ static int ve2_get_hwctx_status(struct amdxdna_client *client, struct amdxdna_dr
 	struct amdxdna_ctx *hwctx;
 	unsigned long hwctx_id;
-	int ret, idx;
+	bool overflow = false;
```
what is meant by overflow, Can u please give proper name

chvamshi-xilinx commented on src/driver/amdxdna/ve2_debug.c:
```diff
@@ -57,24 +56,35 @@ static int ve2_get_hwctx_status(struct amdxdna_client *client, struct amdxdna_dr
-	loc->col = col - hwctx->start_col;
-
-	return hwctx->priv->aie_part;
+        *lead_col = hwctx->start_col;
```
fix the indentation. 
Do we need lead_col ? where are we using it now?

chvamshi-xilinx commented on src/driver/amdxdna/ve2_debug.c:
```diff
@@ -83,42 +93,50 @@ static int ve2_tile_data_reg_write(struct amdxdna_client *client,
-	loc.row = info.row;
-	ret = aie_partition_write(aie_dev, loc, info.addr, sizeof(u32), (void *)&info.val, 0);
-	if (ret < 0) {
+	ret = ve2_partition_write(aie_dev, lead_col, info.col, info.row, info.addr,
```
why we need lead_col in this API? can we simplify?

chvamshi-xilinx commented on src/driver/amdxdna/ve2_debug.c:
```diff
@@ -146,25 +164,28 @@ static int ve2_tile_data_mem_write(struct amdxdna_client *client,
-	loc.row = info.row;
-	ret = aie_partition_write(aie_dev, loc, info.addr, info.size, local_buf, 0);
-	if (ret < 0) {
+	ret = ve2_partition_write(aie_dev, lead_col, info.col, info.row, info.addr,
```
simplify the API

chvamshi-xilinx commented on src/driver/amdxdna/ve2_debug.c:
```diff
@@ -174,34 +195,44 @@ static int ve2_tile_data_reg_read(struct amdxdna_client *client, struct amdxdna_
 
-	loc.row = info.row;
-	ret = aie_partition_read(aie_dev, loc, info.addr, sizeof(u32), (void *)&info.val);
+	ret = ve2_partition_read(aie_dev, lead_col, info.col, info.row, info.addr,
```
lets not add a wrapper if it is calling another API directly, Earlier we required this as we need to support both 6.6 and 6.12

chvamshi-xilinx commented on src/driver/amdxdna/ve2_fw.c:
```diff
@@ -38,14 +35,13 @@ int ve2_store_firmware_version(struct amdxdna_dev_hdl *xdna_hdl, struct device *
 }
 
-static int get_firmware_status(struct amdxdna_dev *xdna, struct device *aie_dev, u32 col)
+static int get_firmware_status(struct amdxdna_dev *xdna, struct device *aie_dev, u32 lead_col, u32 col)
```
Same here, Dont add lead_col here.

chvamshi-xilinx commented on src/driver/amdxdna/ve2_of.c:
```diff
@@ -94,6 +97,12 @@ static int ve2_init(struct amdxdna_dev *xdna)
+		xdna_hdl->hwctx_limit = ve2_hwctx_limit;
+	else
+		xdna_hdl->hwctx_limit = xdna_hdl->priv->hwctx_limit;
+	XDNA_INFO(xdna, "Maximum limit %d hardware context(s)", xdna_hdl->hwctx_limit);
```
fix indentation

chvamshi-xilinx commented on src/driver/amdxdna/ve2_of.h:
```diff
@@ -20,14 +20,68 @@
+	u32			        start_col;
+	u32		        	num_col;
+	struct device		        *aie_dev;
+        struct aie_partition_init_args  *args;
```
fix indentation

chvamshi-xilinx commented on src/driver/amdxdna/ve2_res_solver.h:
```diff
@@ -66,6 +67,8 @@ struct alloc_requests {
 struct xrs_action_load {
 	u32			rid;
 	struct aie_part		part;
+        bool                    create_aie_part;
```
indentation

chvamshi-xilinx commented on src/driver/amdxdna/ve2_res_solver.h:
```diff
@@ -39,6 +39,7 @@ struct aie_qos {
 	u32		latency;	/* Frame response latency */
 	u32		exec_time;	/* Frame execution time */
 	u32		priority;	/* Request priority */
+        u32             exclusive;      /* Exclusive Request or not */
```
indentation

chvamshi-xilinx commented on src/driver/amdxdna/ve2_mgmt.h:
```diff
@@ -41,9 +45,206 @@ struct amdxdna_ctx;
+{
+        u32 offset;
+
+#if KERNEL_VERSION(6, 10, 0) > LINUX_VERSION_CODE
```
We dont need to support <6.12

chvamshi-xilinx commented on src/driver/amdxdna/ve2_mgmt.h:
```diff
@@ -41,9 +45,206 @@ struct amdxdna_ctx;
+ve2_partition_uc_wakeup(struct device *aie_dev, u32 lead_col, u32 col)
+{
+#if KERNEL_VERSION(6, 10, 0) > LINUX_VERSION_CODE
+        col = col + lead_col;
```
Ditto

chvamshi-xilinx commented on src/driver/amdxdna/ve2_mgmt.h:
```diff
@@ -41,9 +45,206 @@ struct amdxdna_ctx;
+		u32 col, u32 row, u32 offset, size_t size, void *buf)
+{
+#if KERNEL_VERSION(6, 10, 0) > LINUX_VERSION_CODE
+	col = col + lead_col;
```
Ditto

chvamshi-xilinx commented on src/driver/amdxdna/ve2_mgmt.h:
```diff
@@ -41,9 +45,206 @@ struct amdxdna_ctx;
+		u32 col, u32 row, u32 offset, size_t size, void *buf)
+{
+#if KERNEL_VERSION(6, 10, 0) > LINUX_VERSION_CODE
+	col = col + lead_col;
```
Ditto

chvamshi-xilinx commented on src/driver/amdxdna/ve2_mgmt.h:
```diff
@@ -41,9 +45,206 @@ struct amdxdna_ctx;
+#if KERNEL_VERSION(6, 10, 0) <= LINUX_VERSION_CODE
+        args->init_opts = (AIE_PART_INIT_OPT_DEFAULT | AIE_PART_INIT_OPT_DIS_TLAST_ERROR) & ~AIE_PART_INIT_OPT_UC_ENB_MEM_PRIV;
+#else
+        args->init_opts = AIE_PART_INIT_OPT_DEFAULT & ~AIE_PART_INIT_OPT_UC_ENB_MEM_PRIV;
```
Ditto

NishadSaraf committed:
Fix return value for aie-partitions query (#766)

Fix return value for aie-partitions query.


(cherry picked from commit 86e76e6eca0176603501016bf0b16c32993b7d41)

Signed-off-by: Nishad Saraf <nishads@amd.com>

bisingha-xilinx committed:
create cma bo with carvedout fashion for ve2 (#748)

Signed-off-by: Bikash Singha <bisingha@xcobisingha40x.xlnx.xilinx.com>
Co-authored-by: Bikash Singha <bisingha@xcobisingha40x.xlnx.xilinx.com>

saifuddin-xilinx committed:
Fixed the review comments

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed the review comments

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

aktondak committed:
[XRT-SMI] Archive migration (#764)

Signed-off-by: Akshay Tondak <aktondak@amd.com>

saifuddin-xilinx committed:
Fixing the codeing style issue

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2_mgmt.c file

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2_mgmt.c file 1

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2_hwctx.c file v1

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2  file v1

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2  file v1

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2  file v2

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed code style issue for ve2  file v3

Signed-off-by: Kaijar, Saifuddin <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed coding style issues

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

maxzhen committed:
fix uninitialized varible (#769)

Signed-off-by: Max Zhen <max.zhen@amd.com>

aktondak committed:
Removing redundant files (#768)

Signed-off-by: Akshay Tondak <aktondak@amd.com>

NishadSaraf committed:
Remove device specific mgmt buffer APIs (#771)

Remove device specific mgmt buffer APIs.

Signed-off-by: Nishad Saraf <nishads@amd.com>

wendyliang25 committed:
test: shim: add read async error multi times in multi threads (#772)

Add test case to read async errors multiple times in multiple
threads.

To test async error read when there is no errors, we will need
to run the test after xdna module is probed and before any runs
launched on the hardware.

Signed-off-by: Wendy Liang <wendy.liang@amd.com>

ManojTakasi committed:
Updated xdna_bo.cpp to maintain bo ref count if we creae on the same process (#774)

Signed-off-by: Manoj Takasi <mtakasi@amd.com>

NishadSaraf committed:
Fix uninitialized app health report pointer (#775)

Fix uninitialized app health report pointer.

Signed-off-by: Nishad Saraf <nishads@amd.com>

NishadSaraf committed:
Free buffer on failure and minor fixes (#776)

Free buffer on failure and minor fixes.

Signed-off-by: Nishad Saraf <nishads@amd.com>

wendyliang25 committed:
test: shim: add instruction code invalid address access test (#773)

Add test case to have instruction code access invalid address
we should expect timeout and then, if we start a good run, the
good run should finish properly.

Signed-off-by: Wendy Liang <wendy.liang@amd.com>

saifuddin-xilinx committed:
Fixed issues after amdxdna_cma memory allocation introduced

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>

saifuddin-xilinx committed:
Fixed review commets

Signed-off-by: Saifuddin Kaijar <saifuddin.kaijar@amd.com>
